### ISSUE TITLE
foreign table data types not converting

### ISSUE THREAD

**Issue by digby2048** [laurenz/oracle_fdw#210]
Hello,

We have a foreign table that isn't converting the data type correctly. The column in the Oracle view is set to RAW. In the postgres foreign table, that column is set to BYTEA which I think is supposed to convert Oracle's RAW to BYTEA in postgres. It doesn't seem to be working correctly.

[HV004] ERROR: column "analyst" of foreign table "act_log" cannot be converted to or from Oracle data type

Should we be using a different data type to convert RAW?
Thanks!

**Comment by laurenz** [comment]
No, `bytea` is the correct type.

Columns are matched by position in oracle_fdw, not by name, so the first column of the foreign table corresponds to the first Oracle column, regardless what the names are. This is often the cause of such problems.

If you can't figure it out yourself, please tell me the statements to define the Oracle table and the PostgreSQL foreign table.

**Comment by digby2048** [comment]
ah ok. I'll check the order of the columns and get back to you. Thanks for the tip!

**Comment by digby2048** [comment]
Yes, that was the problem. I created the foreign table a while ago and I think I had the need to drop it and recreate it. Before I dropped it, I'm guessing I copied the script for it via pgAdmin. I didn't notice that pgAdmin puts the columns in alphabetical order when you view it even though the columns are really in the correct order. I must have run the script again with the columns in alphabetical order which obviously didn't match the order of the Oracle columns. I have recreated the foreign table and it is working now. Thanks so much for your help!
